src/argilla_server/models/database.py

```diff
@@ -352,6 +352,11 @@ def is_draft(self):
     def is_ready(self):
         return self.status == DatasetStatus.ready
 
+    def metadata_property_by_name(self, name: str) -> Optional["MetadataProperty"]:
```

Comment:
Not really sure what's the more pythonic way of defining it but we are using `Union` for this kind of returns?:

```suggestion
    def metadata_property_by_name(self, name: str) -> Union["MetadataProperty", None]:
```